stereo calibration/CalibrationDataCollection.md

```diff
@@ -0,0 +1,69 @@
+- Remember! We are calibrating the cameras for a specific relative position of the cameras, zoom and focus levels!
+- Set the cameras to the smallest aperture
+
+### 1. Design of the checkerboard
+- The pattern should be on a flat and rigid surface 
+
+- White border around it, of roughly the same width as one of the cells (this makes the detection more robust in various environments see [here](https://docs.opencv.org/4.2.0/d9/d0c/group__calib3d.html#gad0e88e13cd3d410870a99927510d7f91))
+
+- Min number of **inner corners**
+    - *Why inner corners?*: "All chessboard related algorithm expects amount of inner corners as board width and height" [^2] - openCV parametrises the width and height based on number of inner corners
+    - 
+
+
+
+- Creating a calibration pattern
+    - [OpenCV](https://docs.opencv.org/4.x/da/d0d/tutorial_camera_calibration_pattern.html)
+    - [Calib.io](https://calib.io/pages/camera-calibration-pattern-generator)
+    - [Mark Hedley Jones blog post](https://markhedleyjones.com/projects/calibration-checkerboard-collection) --- not sure if these print out with a white border!
+
+### 2. Position and orientation of the checkerboard
+For a calibration frame to be useful, the checkerboard must fulfill the following criteria:
+
+- Keep the orientation of the checkboard within 90deg (conservatively 30 deg [^1]) of the "original"/default position
+    - *Why?* The algorithm takes the bottom (right?) corner of the board as the origin, and then traverses the rest of corners from there. If in some frames the board is rotated beyond 30 deg, the origin may be detected at a different physical corner, and all the corners may be traversd in a different order.
+
+- Cover several distances, and within each distance, cover (all parts of the image view)/(of the FOV overlap) [^1]
+    - *Why?*
+    - Note however that if at some distance the calibration pattern is too small, we may not be able to detect the corners reliably and the calibration may be poor. To fix this we may need a larger board (or move cameras closer to the area of interest?) 
+
+- Use a checkerboard as big as possible, ideally with of at least 8x6 squares and a square size of at least 100 mm [^1]
+    - *Why?*
+
+- For a calibration frame to be useful, the checkerboard must be fully visible in both camers
+    - *Why?* Since the algorithm first tries to search for the outer corners of the board, it will fail if the board is only partially visible. This may be avoidable using an alternative algorithm, but the approach [doesn't seem 100% robust](https://github.com/opencv/opencv/issues/15712#issuecomment-1493344373)
+
+- Number of frames captured
+    - From DLC, they claim 30-70 pairs of candidate calibration images should be sufficient, as after corner detection, some of the images might need to be discarded due to either incorrect corner detection or incorrect order of detected corners [^1]
+
+- Keep the pattern steady when aiming to collect a calibration frame
```

Comment:
I assume this means physically holding the checkerboard... not traversing in the same pattern within the FOV to corner 1, then corner 2, than corner 3 and so on?

Reply:
I don't understand this question 🤔

Reply:
For "Keep the pattern steady when aiming to collect a calibration frame", I assume this means maintaining the orientation. For example, one should always hold it in the 8x6 configuration rather than flipping it 90 degrees to also show the 6x8 configuration. 

The other way it could be interpreted is to hold it at the same exact angle, relative to the camera, when walking around the FOV. For example, this would mean holding it perpendicular to the ground always and just walking around, keeping it steady. My understanding is that it's valuble to have a little variation with the degrees of the board relative to the camera, so when standing at one spot, one should 'tilt' the checkerboard like 10degrees to the left and 10degrees to the right. If we keep it perpendicular to the ground at all times, we won't capture minor variations in topography I think? 

Regardless, I am interpreting it as one should keep the checkerboard in the same orientation (like landscape 8x6) for all calibrations

Reply:
Oh I see! I actually just meant "steady" so that the frame is not blurry.

But you are right with the orientation too, as you say we want to keep it in the 8x6 / landscape configuration with small variations in all three angles of rotation (i.e., rotation around the short side of the chessboard, rotation around the long side and rotation around the axis perpendicular to the chessboard)

As you say we want to have these variations in the orientation of the chessboard in all three angles. The reason why we want this is explained a bit more [here](https://aaronolsen.github.io/tutorials/stereomorph/calibration_general.html) (very handwave-ily: it's as if we want to get as close as possible to doing a calibration with a 3D grid of points, but using a checkerboard instead, and so for that we place the board at different orientations)